### ISSUE TITLE
Error: Unable to load the "angular" preset package. Please make sure it's installed.

### ISSUE THREAD

**Issue by jim-moody** [conventional-changelog/conventional-changelog#424]
**Package**
`conventional-recommended-bump`

**Error**
`Error: Unable to load the "angular" preset package. Please make sure it's installed.`

**Description**
Getting this error when using the package as an npm script OR the Node API. The package works when installed globally and run using the CLI.

**Steps to Reproduce**
1. Clone this repo: https://github.com/jim-moody/crb-bug
2. `npm install`
3. `npm run recommend-version`

See the error that displays.

**Comment by stevemao** [comment]
is angular preset installed?

**Comment by jim-moody** [comment]
No, but it wasn't in the previous version, which worked, and it doesn't give any information about having to add presets in the documentation:
https://www.npmjs.com/package/conventional-recommended-bump

**Comment by jim-moody** [comment]
Is that just how it works now? So I guess in this case I would need to install `conventional-changelog-angular`?

**Comment by stevemao** [comment]
I actually thought it was bundled too. @hutson could you confirm?

**Comment by hutson** [comment]
As far as I can tell [from the `package.json` file history](https://github.com/conventional-changelog/conventional-changelog/commits/master/packages/conventional-recommended-bump/package.json), `conventional-changelog-angular` was never a dependency of `conventional-recommended-bump`, nor was `conventional-recommended-bump` setup to use `angular` by default (under the assumption that it was available as a direct dependency).

@stevemao the Angular preset is bundled with `conventional-changelog`, but not `conventional-recommended-bump`.

@jim-moody I'm not sure where else you use `conventional-recommended-bump`, but could this be a case where something else in your dependencies was installing `conventional-changelog-angular` and causing that to get hoisted to the top level `node_modules` directory, right beside `conventional-recommended-bump`, and therefore, it could be loaded correctly by `conventional-recommended-bump`?

**Comment by stevemao** [comment]
I think the preset script used to live in conventional-recommended-bump, then later we refactored it out. However, I don't actually like it to be bundled same reason as https://github.com/conventional-changelog/conventional-changelog/issues/426. Maybe we can add this to the docs?

**Comment by hutson** [comment]
@stevemao I would prefer your approach. Do nothing by default, and document how to install the preferred preset package. Changing to `documentation` label.

**Comment by jim-moody** [comment]
Works for me! Thanks for the information

**Comment by steveoh** [comment]
The way this preset is imported fouls up ncc and esbuild when trying to make a single file build for a github action. Even when the preset is bundled in the output, the loader still cannot find it.

**Comment by vimtaai** [comment]
I'm having the same issue as @steveoh. I cannot use this package as part of a bundled code, it won't find the preset.
It would be nice to have the option to load the preset from code and pass it as an argument instead of the name as a string.

**Comment by dangreen** [comment]
Thanks for the report. The original installation problem was resolved in the discussion: the Angular preset must be installed alongside the package that loads it, and the current documentation covers that setup.

For bundled applications, the current API also supports supplying a custom module loader to `loadPreset()` so the preset can be imported by the bundle instead of being resolved dynamically at runtime. conventional-changelog can also receive an imported preset factory through `.config()`. Closing as completed.
